src/box/lua/load_cfg.lua

```diff
@@ -446,13 +446,15 @@ local log_cfg_option = {
 
 
 local function check_instance_uuid()
-    if box.cfg.instance_uuid ~= box.info.uuid then
+    if box.cfg.instance_uuid ~= box.NULL and
+       box.cfg.instance_uuid ~= box.info.uuid then
         box.error(box.error.RELOAD_CFG, 'instance_uuid')
     end
 end
```

Comment:
Shouldn't we handle `nil` and `box.NULL` differently?

* `nil` -> pass over
* `box.NULL` -> check that `box.cfg.instance_uuid == nil`
* anything else -> compare with `box.info.uuid` (BTW, why not `box.cfg.instance_uuid`?)

Reply:
We don't have the old `box.cfg.instance_uuid` value because it was overwritten by the new one when `box.cfg{instance_uuid = ...}` was executed. I believe this is normal for the options described in `dynamic_cfg` . Because of this, we have no way to compare the old value with the new one.

Reply:
- The option callback isn't invoked if `box.cfg.instance_uuid` is `nil`.
- It'd be complicated because the `box.cfg` option callback currently doesn't have access to the old value. I don't think that fixing it is worthwhile. Semantically, resetting `instance_uuid` to `box.NULL` after it was set to a non-nil value looks correct, which was mentioned in the commit log.
- If the new value stored in `box.cfg.instance_uuid` isn't nil, we compare it with the current value, which is stored in `box.info.uuid`.